### PR TITLE
Added todo list and inbox blocks for canvas

### PR DESCRIPTION
**Pull request by nmghelani**
Block that takes domain and token as input and returns todo list from canvas

### CODE REVIEW

File: GetCanvasTODOList/dcoder_block.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+env: null
+inputs:
+- name: domain
+  type: Text
+  default: sit.instructure.com
+  description: Domain of your institute (ex. sit.instructure.com)
+  required: true
+- name: token
+  type: Text
+  description: Authorization token
```
Comment (ankush-dcoder) [inline GetCanvasTODOList/dcoder_block.yml]: Provide details on how to get this token to help other users.

File: GetCanvasTODOList/dcoder_block.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  required: true
+output:
+  name: todoList
```
Comment (ankush-dcoder) [inline GetCanvasTODOList/dcoder_block.yml:21]: Provide description of output what does it return.

File: GetCanvasTODOList/dcoder_run.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+commands:
+  - run: |
+      # Write your command here
+    default: true
+  - run: |
```
Comment (ankush-dcoder) [inline GetCanvasTODOList/dcoder_run.yml]: Run this command using Dcoder_run or dcoder-cli to create a build in single file in dist folder.

File: GetCanvasTODOList/dcoder_block.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+id: TODO_LIST
+fileName: index.js
```
Comment (ankush-dcoder) [inline GetCanvasTODOList/dcoder_block.yml]: Follow instruction and build into single file and turn off auto install, and update here dist/index.js

Comment (ankush-dcoder) [reply GetCanvasTODOList/dcoder_block.yml]: Read process in the Contributing.md, without build dist blocks are not accepted.
Auto install module blocks are generally slow compare to using a dist build.

File: GetCanvasInbox/dcoder_block.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+id: GET_CANVAS_INBOX
+fileName: index.js
```
Comment (ankush-dcoder) [inline GetCanvasInbox/dcoder_block.yml]: Follow same comment for dist build and use dist/index.js here.

**Review by ankush-dcoder — CHANGES_REQUESTED** [review]
Please use dist build method and complete requested changes. Also make sure to keep auto install off on your public block on Dcoder for these.

**Comment by nmghelani** [comment]
Completed required changes @ankush-dcoder

File: GetCanvasInbox/index.js [hunk 6]
```diff
@@ -2,13 +2,18 @@ const axios = require("axios");
...
   console.log("Fetching data");
   let { domain, token } = inputs;
-  const {data} = await axios.get(
-    `https://${domain}/api/v1/conversations`,
-    {
-      headers: { 'Authorization': `Bearer ${token}` }
-    }
-  );
-  return data;
+  try{
```
Comment (ankush-dcoder) [inline GetCanvasInbox/index.js]: Do not use try catch, let Dcoder's system handle the errors. If you catch the errors, system will not know that block got an error and will try to run flow ahead. No need of try catch here.

File: GetCanvasTODOList/index.js [hunk 7]
```diff
@@ -2,13 +2,19 @@ const axios = require("axios");
...
   console.log("Fetching data");
   let { domain, token } = inputs;
-  const {data} = await axios.get(
-    `https://${domain}/api/v1/users/self/todo`,
-    {
-      headers: { 'Authorization': `Bearer ${token}` }
-    }
-  );
-  return data;
+  try{
```
Comment (ankush-dcoder) [inline GetCanvasTODOList/index.js]: Remove try catch, also remove on Dcoder block, then rebuild and release a new version for both blocks.

**Review by ankush-dcoder — CHANGES_REQUESTED** [review]
Just 2 changes regarding error handling, everything else seems good.

**Comment by nmghelani** [comment]
Removed error handling and published the block to public @ankush-dcoder

**Comment by ankush-dcoder** [comment]
@nmghelani can you get rid of all the semi colons. We do not use semicolons here for clean code.

**Comment by nmghelani** [comment]
> @nmghelani can you get rid of all the semi colons. We do not use semicolons here for clean code.

Done ✅ @ankush-dcoder

File: GetCanvasInbox/index.js [hunk 8]
```diff
@@ -1,12 +1,15 @@
...
-const main = async (inputs, auths, context) => {
-  console.log("Fetching data");
-  let { domain, token } = inputs;
-  const { data } = await axios.get(`https://${domain}/api/v1/conversations`, {
-    headers: { Authorization: `Bearer ${token}` },
-  });
-  console.log("Data fetched successfully");
-  return data;
-};
+const axios = require("axios")
```
Comment (ankush-dcoder) [inline GetCanvasInbox/index.js]: Use single quotes for Strings everywhere.

File: GetCanvasInbox/index.js [hunk 9]
```diff
@@ -1,12 +1,15 @@
...
+const main = async(inputs, auths, context) => {
+  console.log("Fetching data")
```
Comment (ankush-dcoder) [inline GetCanvasInbox/index.js]: Use single quote.

File: GetCanvasInbox/index.js [hunk 10]
```diff
@@ -1,12 +1,15 @@
...
+  let { domain, token } = inputs
+  const {data} = await axios.get(
+    `https://${domain}/api/v1/conversations`,
+    {
+      headers: { 'Authorization': `Bearer ${token}` }     
+    }
+  )
+  console.log("Data fetched successfully")
```
Comment (ankush-dcoder) [inline GetCanvasInbox/index.js]: Use single quote.

**Review by ankush-dcoder — CHANGES_REQUESTED** [review]
Use single quotes for strings for consistency with repo logic.

**Pull request merged**